src/app/loadout/mod-utils.ts

```diff
@@ -120,20 +120,18 @@ export function getItemEnergyType(
   }
 }
 
+function getItemTypeOrTierDisplayName(plugDef: PluggableInventoryItemDefinition): string {
+  // XXX: Class Item Artifact Mods are labeled "Class Item Mod" instead of "Class Item Armor Mod"
+  if (plugDef.itemTypeDisplayName === 'Class Item Mod') {
```

Comment:
Unfortunately this will only work in the English manifest. We need to operate on some field that isn't translated, or generate a table in d2-additional-info to use.

There are some helper methods for getting extended socket details which may be useful for this.

Reply:
I'll investigate, thanks for getting back to me!